### PR TITLE
Initial implementation of a android quick start application

### PR DESCRIPTION
**Pull request by thehoneymad**
Wrote an initial implementation of a android quick start application.
This is a standalone project built with gradle. Dependencies include:
1. Maplibre
2. Amplify Geo Android
3. Amplify Geo MapLibre Adaptor

The app has a basic Amplify Map Control with Cognito. It uses Amplify Geo's
MapLibreView over native MapLibre MapView. It has cognito userpool
authentication integrated.

I also implemented a demo place picker using SearchPlaceIndexForCoordiantes API.
I used basic android ImageView and TextView for quickstart. Code quality
is scrappy because this is supposed to be a quickstart app.

This commit is buildable by its own. Adds the place picker sample with a single
commit.

test: gradle build

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Hi @thehoneymad, thanks for contributing a sample.

Please be patient while I look for someone among the maintainers who knows Android more than me for a quick review.

**Comment by thehoneymad** [comment]
@dreamorosi sounds good mate!

File: android-quick-start/README.md
Comment (eeatonaws) [inline android-quick-start/README.md:13]:
```suggestion
1. An Amazon Location map resource
```

File: android-quick-start/README.md
Comment (eeatonaws) [inline android-quick-start/README.md:14]:
```suggestion
1. An Amazon Location place-index resource
```

File: android-quick-start/README.md
Comment (eeatonaws) [inline android-quick-start/README.md:23]:
```suggestion
This project uses a JSON configuration following Amplify's configuration schema in `app/src/main/res/raw/amplifyconfiguration.json`. To use the quick start, we need to replace the following in the configuration JSON.
```

File: android-quick-start/app/src/main/java/com/amazon/location/android/quickstart/MainActivity.kt
Comment (eeatonaws) [inline android-quick-start/app/src/main/java/com/amazon/location/android/quickstart/MainActivity.kt:35]: Since Amplify can only be configured once at runtime, it's best to add the plugins and configure Amplify in an application class to avoid edge cases where the onCreate for an activity might be called more than once.

**Review by eeatonaws — COMMENTED** [review]
Left a few comments, but overall looks good.

File: android-quick-start/app/src/main/java/com/amazon/location/android/quickstart/MainActivity.kt
Comment (thehoneymad) [reply android-quick-start/app/src/main/java/com/amazon/location/android/quickstart/MainActivity.kt:35]: That's a good callout. I will move the initialization to an Application class.

File: android-quick-start/README.md
Comment (thehoneymad) [reply android-quick-start/README.md:23]: Ah, thank you for catching that. I will take this suggestion.

File: android-quick-start/README.md
Comment (dreamorosi) [inline android-quick-start/README.md:15]:
```suggestion
1. A Cognito Identity Pool for unauthorized access
```

File: android-quick-start/README.md
Comment (dreamorosi) [inline android-quick-start/README.md:19]: 1. Configuring Amplify to set map, place-index, and Cognito Identity Pool id.

File: android-quick-start/README.md
Comment (dreamorosi) [inline android-quick-start/README.md:29]: For example, if we are building this application for region `us-west-2`, map resource name `sample-map`, map style name `VectorEsriStreets`, place-index resource name `sample-place-index`, and Identity Pool id `us-west-2:cognito-pool-id`, the `amplifyconfiguration.json` will look like the following:

**Review by dreamorosi — APPROVED** [review]
Thanks @eeatonaws for the review!

@thehoneymad left only a couple of very minor comments on service naming.

Looking forward to merge this PR when all comments have been addressed or discussed.

**Comment by thehoneymad** [comment]
Appreciate all the nice comments @eeatonaws. I will up a new revision soon.

File: android-quick-start/README.md
Comment (thehoneymad) [reply android-quick-start/README.md:13]: Made this change.

File: android-quick-start/README.md
Comment (thehoneymad) [reply android-quick-start/README.md:14]: Made this one too.

File: android-quick-start/app/src/main/java/com/amazon/location/android/quickstart/MainActivity.kt
Comment (thehoneymad) [reply android-quick-start/app/src/main/java/com/amazon/location/android/quickstart/MainActivity.kt:35]: I have moved initialization of Amplify to a dedicated class named `AndroidQuickstartApp` :)

File: android-quick-start/README.md
Comment (thehoneymad) [reply android-quick-start/README.md:15]: Made this change.

**Comment by thehoneymad** [comment]
@dreamorosi , @eeatonaws I updated the PR with a new change addressing the review comments.

File: android-quick-start/README.md
Comment (thehoneymad) [reply android-quick-start/README.md]: I made this change

File: android-quick-start/README.md
Comment (thehoneymad) [reply android-quick-start/README.md]: I also made this change.

**Review by eeatonaws — APPROVED** [review]
Thanks for making the changes, looks great!

**Pull request merged**
